### PR TITLE
Add g:transparent_groups and g:transparent_my_tp

### PR DESCRIPTION
**Pull request by JohanChane**
I like the transparent color scheme, so i use this plugins to "diy" a transparent color scheme.

### CODE REVIEW

**Comment by tribela** [comment]
Thank you for loving this plugin and submitting a PR.

But removing default group is breaking change and getting confused for light users.
The basic configuration should be just adding plugin without configuration.

And I don't know what `g:transparent_my_tp` is for.

Could you make original transparent group as default but customizable, And explain what `g:transparent_my_tp` is?

**Comment by JohanChane** [comment]
Just for executing highlight command to set foreground or background. It is better to delete it, and use the below:

```vim
augroup transparent
    autocmd VimEnter,ColorScheme * call MyTransparent()
augroup END

function! MyTransparent()
    " CursorLine
    "hi CursorLine ctermfg=NONE ctermbg=239 guibg=NONE guibg=#4e4e4e

    " coc.nvim
    "hi CocMenuSel ctermbg=239 guibg=#4e4e4e
endfunction
```

I think it is good to add `g:transparent_groups`. For example, Many highlight groups link the `CursorLine` group. e.g. `CocListLine`, `g:fzf_colors.bg+` etc. If I don't like ctermbg and guibg of the `CursorLine`, I can add `CursorLine` to `g:transparent_groups`. At the same time, if I don't like the ctermfg and guifg, I can config them in `MyTransparent` function.<br>
On the other hand, some plugins use itself highlight groups like `CocFloating`. So in case of confusing the light users, add the minimal configuration.

File: autoload/background.vim [hunk 1]
```diff
@@ -39,10 +39,7 @@ endfunction
 
 function! background#clear_background()
     if g:clear_background
-        for group in ['Normal', 'Comment', 'Constant', 'Special', 'Identifier',
-                    \'Statement', 'PreProc', 'Type', 'Underlined', 'Todo', 'String',
-                    \'Function', 'Conditional', 'Repeat', 'Operator', 'Structure',
-                    \'LineNr', 'NonText', 'SignColumn', 'CursorLineNr', 'EndOfBuffer']
+        for group in g:transparent_groups
```
Comment (tribela) [inline autoload/background.vim:42]: Set default groups to not breaking existing users
```suggestion
        if !exists('g:transparent_groups')
            let g:transparent_groups = 
                        \['Normal', 'Comment', 'Constant', 'Special', 'Identifier',
                        \'Statement', 'PreProc', 'Type', 'Underlined', 'Todo', 'String',
                        \'Function', 'Conditional', 'Repeat', 'Operator', 'Structure',
                        \'LineNr', 'NonText', 'SignColumn', 'CursorLineNr', 'EndOfBuffer']
        endif
        for group in g:transparent_groups
```

File: README.md [hunk 2]
```diff
@@ -28,6 +28,163 @@ Plug 'tribela/vim-transparent'
...
 
+## Configuration
+
+### Minimal configuration
+
+```vim
+Plug 'tribela/vim-transparent'
+
+" Clear the background of highlight groups in g:transparent_groups.
+let g:transparent_groups = ['Normal', 'Comment', 'Constant', 'Special', 'Identifier',
```
Comment (tribela) [inline README.md:39]: As default values are implemented, This config is not needed on minimal configuration

File: README.md [hunk 3]
```diff
@@ -28,6 +28,163 @@ Plug 'tribela/vim-transparent'
...
+                            \ 'Statement', 'PreProc', 'Type', 'Underlined', 'Todo', 'String',
+                            \ 'Function', 'Conditional', 'Repeat', 'Operator', 'Structure',
+                            \ 'LineNr', 'NonText', 'SignColumn', 'CursorLineNr', 'EndOfBuffer']
```
Comment (tribela) [inline README.md:42]: As default values are set, This should not be set on minimal configuration
```suggestion
```

Comment (JohanChane) [reply README.md]: I have deleted the `Minimal Configuration` and commited.

**Pull request merged**